### ISSUE TITLE
Use of oblivious types when using generic type constraint 'class'

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn-analyzers#4040]
### Describe the bug

I'm not 100% sure this is a bug, but I can't see how to improve it.

### C.cs

```c#
#nullable enable

public class C<T>
//           ^ RS0041 Symbol 'C<T>' uses some oblivious reference types.
    where T : class
{
}

```

### PublicAPI.Unshipped.txt

```
#nullable enable
C<T>.C() -> void
C<T>
```

### Expected behavior

No diagnostic.

### Actual behavior

Analyzer flagging `C<T>` as using some oblivious types.

The diagnostic goes away when the constraint is removed.

### Analyzer

**Package**: [Microsoft.CodeAnalysis.PublicApiAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers)

**Version**: v3.3.0 (Latest)

**Diagnostic ID**: [RS0041](https://docs.microsoft.com/visualstudio/code-quality/rs0041)

**Comment by mavasani** [comment]
@jcouv

**Comment by Evangelink** [comment]
@jcouv Do you need some help moving forward with this ticket? Honestly, I am a little lost with the idea of this rule but if you do a little briefing of the current scope of the rule and what needs to be changed I would be happy to help you.

**Comment by jcouv** [comment]
Thanks for the ping. I'll look at it this week.
